File: Algorithms/Python/1 Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        m = {}
        for i, v in enumerate(nums):
            if target - v in m:
                result.append(i)
                result.append(m[target - v])
                # assume only one solution
                break
            else:
                m[v] = i
        
        return result

File: Algorithms/CPP/1 Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
	    unordered_map<int, int> map;
	    vector<int> result;
	    for (int i = 0; i < nums.size(); ++i) {
		    if (map.find(target - nums[i]) != map.end()) {
			    result.push_back(i);
			    result.push_back(map[target - nums[i]]);
          # assume only one solution
          break;
		    }
		    else {
			    map[nums[i]] = i;
		    }
	    }

	    return result;
    }
};
